### PR TITLE
Move Android CI to Buildkite and switch to clustered setup

### PR DESCRIPTION
**Pull request by mokagio**
This PR combines:

- #6320 
- #6471 
- Updates the iOS pipelines to use Xcode ~~15.1~~ 15.0.1, which, combined with and internal PR, is all we need to run on the "clustered" Buildkite infrastructure. Aside from the fact that our Xcode 15.1 runs faster, we _need_ the clustered setup for the Android tests to run properly.

Notice I created this PR with "[Status] DO NOT MERGE" to port over @oguzkocer's https://github.com/wordpress-mobile/gutenberg-mobile/pull/6320#discussion_r1430442650 and warning about merging these changes only after the updates to the Android Buildkite infrastructure have been properly deployed.

---

Update: I don't know why yet, but the Xcode 15.1 CI failed where the Xcode 15.0.1 didn't. I actually made two commits that represented different experiments, so I'll need to go back and check which setup had the issue. Still, CI is green now and this can be looked over once more for safety, then merged.

- https://github.com/wordpress-mobile/gutenberg-mobile/pull/6521/commits/181097345f0d64dc6622db4c3c0512cd8d653a63
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/6521/commits/ced0ec072611a9dda04e503c65bb754dcd9c92de

### CODE REVIEW

**Comment by mokagio** [comment]
https://github.com/wordpress-mobile/gutenberg-mobile/pull/6525 addresses the issue encountered in CI for the iOS builds. 

Once that lands on `trunk`, a rebase should be all we need to see a whole green build on this branch.

**Comment by mokagio** [comment]
Apple Silicon CI failing to build, possibly due to 

```
ld: warning: search path '/Applications/Xcode-15.1.0.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.2.sdk/usr/lib/swift/opt/ci/builds/builder/automattic/gutenberg-mobile/gutenberg/packages/react-native-editor/ios' not found
```

Investigating

**Comment by mokagio** [comment]
Good news, I can replicate the issue locally using 

```
export RN_EDITOR_E2E_IOS_DESTINATION="platform=iOS Simulator,name=iPhone 15 Pro"
npm run test:e2e:bundle:ios && npm run core test:e2e:build-app:ios
```

**Review by geriux — APPROVED** [review]
LGTM! Nice work @mokagio 👏 All CI checks are green 🚀

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ x-common-params:
   - &ci_toolkit_plugin
     automattic/a8c-ci-toolkit#2.18.2
   - &xcode_agent_env
-    IMAGE_ID: xcode-15.1
+    IMAGE_ID: xcode-15.0.1
```
Comment (geriux) [inline .buildkite/pipeline.yml:33]: One thing I've just realized is that this version changed, is it expected? Thanks!

Comment (mokagio) [reply .buildkite/pipeline.yml:33]: Hey @geriux ! Thank you for checking.

Yes. It's expected.

I don't know why yet, but the Xcode 15.1 CI failed where the Xcode 15.0.1 didn't. I actually made two commits that represented different experiments, so I'll need to go back and check which setup had the issue.

https://github.com/wordpress-mobile/gutenberg-mobile/commit/181097345f0d64dc6622db4c3c0512cd8d653a63
https://github.com/wordpress-mobile/gutenberg-mobile/commit/ced0ec072611a9dda04e503c65bb754dcd9c92de

Comment (geriux) [reply .buildkite/pipeline.yml:33]: Sounds good!

**Pull request merged**